src/librustc/hir/map/mod.rs

```diff
@@ -995,6 +997,7 @@ impl<'hir> Map<'hir> {
     pub fn span(&self, hir_id: HirId) -> Span {
         self.read(hir_id); // reveals span from node
         match self.find_entry(hir_id).map(|entry| entry.node) {
+            Some(Node::Arg(arg)) => arg.pat.span,
```

Comment:
This should be `arg.span` now

Reply:
In this new commit, is the `print_arg` function too simple? Maybe it isn't necessary?